examples/stm32/nucleo-f746zg-baremetal/Makefile

```diff
@@ -37,6 +37,15 @@ cmsis_core:     # ARM CMSIS core headers
 	git clone --depth 1 -b 5.9.0 https://github.com/ARM-software/CMSIS_5 $@
 cmsis_f7:       # ST CMSIS headers for STM32F7 series
 	git clone --depth 1 -b v1.2.8 https://github.com/STMicroelectronics/cmsis_device_f7 $@
+mbedtls:        # mbedTLS library
+	git clone --depth 1 -b v2.28.2 https://github.com/mbed-tls/mbedtls $@
+
+ifeq ($(TLS),1)
+CFLAGS += -DMG_ENABLE_MBEDTLS=1 -DMBEDTLS_CONFIG_FILE=\"mbedtls_config.h\"
+CFLAGS += -Wno-conversion -Imbedtls/include
+SOURCES += mbedtls/library/*.c
+firmware.elf: mbedtls
+endif
```

Comment:
We are instructing to use CFLAGS_EXTRA in OS examples. I would do the same here, though the user would just add:
`CFLAGS_EXTRA=-DMG_ENABLE_MBEDTLS=1` (with of course no linking stuff)
This will keep the examples the same in both OS and embedded worlds; we cant catch `CFLAGS_EXTRA` existence and being `==-DMG_ENABLE_MBEDTLS=1` to add extra flags/sources

Reply:
For the pre-built libs, using CFLAGS_EXTRA is probably fine. Cause this is really just CFLAGS_EXTRA

When we're pulling an extra repo and adding extra sources, it is not really extra CFLAGS anymore